attorch/train.py

```diff
@@ -31,17 +31,25 @@ def early_stopping(model, objective, interval=5, patience=20, start=0, max_iter=
         max_iter:  maximum number of iterations before the iterator terminated
         maximize:  whether the objective is maximized of minimized
     """
+    training_status = model.training
+
+    def _objective(mod):
```

Comment:
Actually an easy solution would be to have an optional argument `switch_mode=True` that controls whether more switching will be attempted. If you like prefer the early stopping to not touch the `model`'s training mode, you can set it to `False` 🙂